### PR TITLE
JT-40

### PR DESCRIPTION
**Pull request by fstk1337**

### CODE REVIEW

File: import/src/main/java/saffchen/database/GSheetConnection.java [hunk 1]
```diff
@@ -16,38 +20,52 @@
 
 public class GSheetConnection {
 
-    private static Properties properties = new Properties();
+    private static final Properties properties = PropertiesLoader.load("gsheet.properties");
```
Comment (pelmegov) [inline import/src/main/java/saffchen/database/GSheetConnection.java:23]: Does it make sense to move inside application.properties and inject here like GsheetProperties autowired object or something like that?

File: import/src/main/java/saffchen/utils/GSheetImportUtils.java [hunk 2]
```diff
@@ -13,59 +13,54 @@
 
 
 public class GSheetImportUtils implements ImportUtils {
-    private final String RANGE = GSheetConnection.RANGE;
-    private Sheets service;
+    private final Sheets service;
 
     public GSheetImportUtils(Sheets service) {
```
Comment (pelmegov) [inline import/src/main/java/saffchen/utils/GSheetImportUtils.java:18]: Minor: lombok.

File: import/src/main/java/saffchen/ImportFromGSheetCommand.java [hunk 3]
```diff
@@ -31,23 +27,12 @@ public void doCommand() {
...
 
             GSheetImportUtils gSheetImportUtils = new GSheetImportUtils(GSheetConnection.getSheetsService());
-            fileStorageUtils.addRawProductsFromListToCSV(gSheetImportUtils.checkTheDuplicates(
+            String result = fileStorageUtils.addRawProductsFromListToCSV(gSheetImportUtils.checkTheDuplicates(
                     gSheetImportUtils.getData(),
                     fileStorageUtils.getDataFromCSV()));
-        } catch (GeneralSecurityException e) {
-            e.printStackTrace();
-        } catch (IOException e) {
+            System.out.println(result);
```
Comment (pelmegov) [inline import/src/main/java/saffchen/ImportFromGSheetCommand.java:33]: Let's use logger ( @Slf4j probably ) instead of printing to console.

Comment (pelmegov) [reply import/src/main/java/saffchen/ImportFromGSheetCommand.java:33]: Actually logger already in this class.

File: import/src/main/java/saffchen/utils/PropertiesLoader.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertiesLoader {
+    public static Properties load(String fileName) {
+        Properties properties = new Properties();
+        try (InputStream in = PropertiesLoader.class.getClassLoader().getResourceAsStream(fileName)) {
+            properties.load(in);
+        } catch (IOException e) {
+            System.out.println("Can't load properties from file " + fileName);
```
Comment (pelmegov) [inline import/src/main/java/saffchen/utils/PropertiesLoader.java:13]: Logger.

File: import/src/main/java/saffchen/utils/PropertiesLoader.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package saffchen.utils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertiesLoader {
+    public static Properties load(String fileName) {
```
Comment (pelmegov) [inline import/src/main/java/saffchen/utils/PropertiesLoader.java:8]: Do we need this property loader at all?

File: import/src/main/java/saffchen/utils/GSheetImportUtils.java [hunk 6]
```diff
@@ -13,59 +13,54 @@
...
         this.service = service;
     }
 
     @Override
     public List<RawProduct> getData() {
-        ValueRange result = null;
-        List<List<String>> values = null;
-        List<String> listOfStrProducts = new ArrayList<>();
-        List<RawProduct> products = new ArrayList<>();
+        List<List<String>> values = getRowsFromGSheet(service);
```
[the remark this answers is not in the document]
Comment (fstk1337) [reply import/src/main/java/saffchen/utils/GSheetImportUtils.java]: There's no need. Argument removed.

File: import/src/main/java/saffchen/utils/GSheetImportUtils.java (hunk 2 above)
Comment (fstk1337) [reply import/src/main/java/saffchen/utils/GSheetImportUtils.java]: Lombok added.

File: import/src/main/java/saffchen/ImportFromGSheetCommand.java (hunk 3 above)
Comment (fstk1337) [reply import/src/main/java/saffchen/ImportFromGSheetCommand.java]: Okay, using logger.

File: import/src/main/java/saffchen/utils/PropertiesLoader.java (hunk 4 above)
Comment (fstk1337) [reply import/src/main/java/saffchen/utils/PropertiesLoader.java]: The class was removed.

File: import/src/main/java/saffchen/utils/PropertiesLoader.java (hunk 5 above)
Comment (fstk1337) [reply import/src/main/java/saffchen/utils/PropertiesLoader.java]: The class was removed.

**Review by pelmegov — APPROVED** [review]

**Pull request merged**